umbraco/Umbraco-CMS
# Fix tracked reference queries for SqlServer

Fixes https://github.com/umbraco/Umbraco-CMS/issues/15866

### Summary
Fixed queries to work for both SqlServer and Sqlite.
Further, optimised a bit, so the actual paged query is not executed if total is 0.

Also added a test to this use case.

###  Test 
- Setup a doc type with content picker
- Create two pages of doc type and from one of then reference the other
  - Verify the info tab to do not anything on the page that have the other item picked
  - Verify it shows up on the info tab on the referenced item.
